internal/impl/devbox.go

```diff
@@ -712,9 +700,12 @@ func (d *Devbox) computeNixEnv(ctx context.Context, usePrintDevEnvCache bool) (m
 	debug.Log("current environment PATH is: %s", currentEnvPath)
 	// Use the original path, if available. If not available, set it for future calls.
 	// See https://github.com/jetpack-io/devbox/issues/687
-	originalPath, ok := env["DEVBOX_OG_PATH"]
+	// We add the project dir hash to ensure that we don't have conflicts
+	// between different projects (including global)
+	// (moving a project would change the hash and that's fine)
+	originalPath, ok := env["DEVBOX_OG_PATH_"+d.projectDirHash()]
 	if !ok {
-		env["DEVBOX_OG_PATH"] = currentEnvPath
+		env["DEVBOX_OG_PATH_"+d.projectDirHash()] = currentEnvPath
```

Comment:
this feels like an important but easy-to-overlook-or-break scenario. Could we augment `testscripts/shell/shellenv.test.txt` to test for the `devbox rm` scenario mentioned in the issue? cc @ipince

Reply:
For some reason `devbox` gets removed from path in `testscripts/shell/shellenv.test.txt` (not sure if sourcing a new path is doing it and somehow the path that gets used by devbox is different than the path in testscripts)

So instead, wrote a few unit tests directly.